mdpow/tests/test_automated_dihedral_analysis.py

```diff
@@ -206,26 +265,22 @@ def test_save_fig_info(self, SM25_tmp_dir, caplog):
         caplog.clear()
         caplog.set_level(logging.INFO, logger='mdpow.workflows.dihedrals')
         dihedrals.automated_dihedral_analysis(dirname=SM25_tmp_dir, figdir=SM25_tmp_dir,
-                                              resname=self.resname, molname='SM25',
+                                              resname=resname, molname='SM25',
                                               solvents=('water',))
         assert f'Figure saved as {SM25_tmp_dir}/SM25/SM25_C10-C5-S4-O11_violins.pdf' in caplog.text, 'PDF file not saved'
 
-    def test_DataFrame_input(self, SM25_tmp_dir):
-        test_df = pd.DataFrame([['C1-C2-C3-C4', 'water', 'Coulomb', 0, 0, 60.0],
-                                ['C1-C2-C3-C5', 'water', 'Coulomb', 0, 0, 60.0]],
-                                [1,2],['selection', 'solvent', 'interaction', 'lambda', 'time', 'dihedral'])
-        plot = dihedrals.automated_dihedral_analysis(dirname=SM25_tmp_dir, figdir=SM25_tmp_dir,
-                                                     resname=self.resname,
-                                                     solvents=('water',), dataframe=test_df)
-        assert isinstance(plot, seaborn.axisgrid.FacetGrid)
+    def test_DataFrame_input(self, SM25_tmp_dir, dihedral_data):
+        df, _ = dihedral_data
+        dihedrals.automated_dihedral_analysis(dirname=SM25_tmp_dir, figdir=SM25_tmp_dir,
+                                              resname=resname, molname=molname,
+                                              solvents=('water',), dataframe=df)
```

Comment:
This takes a while to run and is run in multiple tests as far as I can tell. Can we make it a module-level fixture and only run it once?

Reply:
We can do two fixtures for now with `scope=class` if you think it will reduce runtime. There are two distinct cases, each used twice, for four tests total. One case is the regular analysis and the other tests analysis with DataFrame input.

Reply:
cutting run time in half is already good, please do!

Reply:
@orbeckst my temporary testing directory uses `tmp_path` which is function scoped according to pytest. Since most other downstream tests call that fixture, they cannot exceed the function scope. Can we raise an issue to change to `tmp_dir_factory` in lieu of making these changes now?

Reply:
#235

Reply:
ok, add a comment to the current tests that they should be changed and reference #235